lib/elixir/lib/dynamic_supervisor.ex

```diff
@@ -942,8 +942,9 @@ defmodule DynamicSupervisor do
 
   defp restart_child(:one_for_one, current_pid, child, state) do
     {{m, f, args} = mfa, restart, shutdown, type, modules} = child
+    %{extra_arguments: extra} = state
 
-    case start_child(m, f, args) do
+    case start_child(m, f, extra ++ args) do
```

Comment:
If we generate a `:start_error` error report the child will not include the extra arguments in the report. We need to add those arguments in the report. In fact I think we have this bug ever time we report an error! As the extra args are always the same I think we can fetch it from the state in the reporting function, rather than than edit the child in the report.

Reply:
This can be done in another PR :D

Reply:
I looked into it and wrote a fix (but no test yet) for the error reporting as well. I can push it here or create another PR, whatever you prefer.

Reply:
Please do send a new PR as I am merging this one. You can add a test in the translator_test file. Here is the last time we added a DynamicSupervisor test to it: https://github.com/elixir-lang/elixir/commit/f0ad7edf411de74fb0d2cf4384580a2e0672b01f